### ISSUE TITLE
How to read non-binary head file?

### ISSUE THREAD

**Issue by pinshuai** [modflowpy/flopy#1090]
Hi, I was trying to load a modflow file (mf2k) using flopy `hds = flopy.utils.HeadFile(head_file, model=ml)`, but I got the following error: 

```
/opt/anaconda3/envs/modflow/lib/python3.9/site-packages/flopy/utils/binaryfile.py:308: UserWarning: Possible error. ncol (808595763) * nrow (808142128) > 10,000,000 
  warnings.warn(s)
```
I can open the .hds file in a text editor so I think it is a non-binary file.

Do you know any function within Flopy that can read a non-binary output?

**Comment by spaulins-usgs** [comment]
Try using FormattedHeadFile:

hds = flopy.utils.FormattedHeadFile(head_file)

I believe FormattedHeadFile was written for mf2005 output files, but it may work with mf2k files too.

**Comment by pinshuai** [comment]
I tried using `hds = flopy.utils.FormattedHeadFile(head_file)`, and it seemed successful. I can view the header using

```
> hds.nrow, hds.ncol, hds.nlay, hds.times
(600, 600, 24, [15312.0])
```

However when I tried reading the data using `hds.get_data()`, I got the following error:

```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-12-acdbf8364a37> in <module>
----> 1 hds.get_data()

/opt/anaconda3/envs/modflow/lib/python3.9/site-packages/flopy/utils/datafile.py in get_data(self, kstpkper, idx, totim, mflay)
    537             totim1 = self.times[-1]
    538 
--> 539         data = self._get_data_array(totim1)
    540         if mflay is None:
    541             return data

/opt/anaconda3/envs/modflow/lib/python3.9/site-packages/flopy/utils/datafile.py in _get_data_array(self, totim)
    453             ncol = self.recordarray["ncol"][idx]
    454             shp = (nrow, ncol)
--> 455             data[ilay - 1] = self._read_data(shp)
    456         return data
    457 

/opt/anaconda3/envs/modflow/lib/python3.9/site-packages/flopy/utils/formattedfile.py in _read_data(self, shp)
    203                 if not is_float(val):
    204                     raise Exception(
--> 205                         "Invalid data encountered while reading data file."
    206                         + " Unable to convert data to float."
    207                     )

Exception: Invalid data encountered while reading data file. Unable to convert data to float.
```

**Comment by spaulins-usgs** [comment]
MF2000 formatted file format must be a different format than MF2005.

@langevin-usgs Do we want to support MF2000 for reading heads out of a formatted file?

**Comment by langevin-usgs** [comment]
I'm guessing that @pinshuai could adjust the CHEDFM value and/or experiment with the LABEL option and flopy would read the ascii heads.  The following is from the online guide to modflow-2000.  Hopefully this helps.

![image](https://user-images.githubusercontent.com/3372403/113737863-05ccdb00-96c4-11eb-8356-06f502d2cad6.png)

**Comment by pinshuai** [comment]
Unfortunately I am reading an archived modflow 2000 model output files, so I cannot rerun the model using the different head save format options. 

If the head is in an ASCII file format, it should be easy for flopy to read, right? The header looks like something below:

```
     1  504   1.000000E+00   1.531200E+04             HEAD   600   600     1 (20F10.3)  
```
It's just a 600x600x504 matrix. I can provide the .hds file if you want to take a look.

**Comment by langevin-usgs** [comment]
It might be easier for you to just use some regular old python line reading and parsing in order to read that head file.  It shouldn't be too hard to do.  Here is a simple function in flopy that you might be able to adapt for your needs.  Good luck!

```
def read1d(f, a):
    """
    Quick file to array reader for reading gridgen output.  Much faster
    than the read1d function in util_array

    """
    dtype = a.dtype.type
    lines = f.readlines()
    l = []
    for line in lines:
        l += [dtype(i) for i in line.strip().split()]
    a[:] = np.array(l, dtype=dtype)
    return a
```

**Comment by pinshuai** [comment]
@langevin-usgs thanks for your quick reply. The head file turned out to be the culprit. As I was trying to read the file using pd.read_csv(), I found there was a bad data point `456.11-1653.21` (note the hyphen in between) in the file that was causing the issue. After I fixed that one data, flopy is now able to read the entire head file. I guess I should blame the person who have prepared the data package 😞
